juju/juju
# agent: set state password with serving info

## Description of change

When an agent config has state serving info,
ensure it also has a state password set by
copying it from the API details.

Also, don't accept state addresses, as they're
not used. We'll keep them in the on-disk
format for reading in in case the stateport
attribute is missing, in which case there
is code to deduce from the state addresses.

The deployer facade's StateAddresses method
has been removed. This was done without
bumping the facade or retaining backwards
compatibility, since no version from 2.0 on
ever used that method.

## QA steps

1. juju bootstrap localhost
2. lxc launch ubuntu:xenial x1
3. lxc launch ubuntu:xenial x2
4. juju switch controller
5. juju add-machine ssh:x1.lxd
6. juju add-machine ssh:x2.lxd
7. juju enable-ha --to 1,2
8. juju debug-log
(check that there are no panics; additional controllers won't come online as they do not initialise the replicaset)

## Documentation changes

None

## Bug reference

Partially fixes https://bugs.launchpad.net/juju/+bug/1642618
